src/tree.rs

```diff
@@ -1141,6 +1145,32 @@ fn detect_cycles(parents: &[ResolvedParent]) -> Option<Index> {
     None
 }
 
+fn break_cycles(builder: &mut Builder, cycle_start_index: Index) -> Result<()> {
+    let mut unfiled = Item::new(UNFILED_GUID, Kind::Folder);
+    unfiled.age = 0;
+    unfiled.needs_merge = true;
+
+    let cycle_start_guid = builder
+        .entries[cycle_start_index]
+        .item
+        .clone()
+        .guid;
+
+    builder
+        .item(unfiled)?;
+    //     .by_parent_guid(ROOT_GUID)?;
+    // &mut*builder
+    //     .parent_for(&UNFILED_GUID)
+    //     .by_children(&ROOT_GUID)?;
```

Comment:
From the resulting structure I'm getting, I'm assuming I'm referencing Root incorrectly, and or setting root as unfiled's parent. Is this right? I copy-pasted from [trees.md](https://github.com/mozilla/dogear/blob/master/docs/trees.md) but maybe there's a better (read right) way to do this.

Reply:
For instance, the builder that results from these changes has the parent of root as Root. But when lines 1161 - 1164 are uncommented this is the parent object for unfiled: ```parent: Partial(
                [
                    UnknownItem(
                        Guid("root________"),
                    ),
                    Children(
                        0,
                    ),
                ],
            ),```